activerecord/test/models/developer.rb

```diff
@@ -294,3 +294,9 @@ class AttributedDeveloper < ActiveRecord::Base
 
   self.ignored_columns += ["name"]
 end
+
+class ColumnNamesCachedDeveloper < ActiveRecord::Base
+  self.table_name = "developers"
+  attribute :name, DeveloperName.new
```

Comment:
This line isn't necessary to reproduce the problem, so including it makes the test less clear.

```suggestion

```